messaging/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsMessage.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c)  2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.messaging.connectors.jms;
+
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.jms.BytesMessage;
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSException;
+import javax.jms.MessageEOFException;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import io.helidon.messaging.MessagingException;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+/**
+ * Message representing JMS message together with all the metadata.
+ *
+ * @param <T> Type of the payload.
+ */
+public interface JmsMessage<T> extends Message<T> {
+
+    /**
+     * The logger.
+     */
+    Logger LOGGER = Logger.getLogger(JmsMessage.class.getName());
+
+    /**
+     * Original JMS message received.
+     *
+     * @return original JMS message received
+     */
+    javax.jms.Message getJmsMessage();
+
+    /**
+     * Metadata about the JMS session.
+     *
+     * @return JMS session
+     */
+    Session getJmsSession();
+
+    /**
+     * Get client's connection to its JMS provider.
+     *
+     * @return JMS connection
+     */
+    Connection getJmsConnection();
+
+    /**
+     * Get JMS connection factory.
+     *
+     * @return JMS connection factory
+     */
+    ConnectionFactory getJmsConnectionFactory();
+
+    /**
+     * Check if message has been acknowledged yet.
+     *
+     * @return true if message has been acknowledged
+     */
+    boolean isAck();
+
+    /**
+     * Create a message with the given payload.
+     *
+     * @param msg          JMS message to be wrapped
+     * @param executor     Executor used for invoking ack
+     * @param sessionEntry metadata about the JMS session
+     * @return A message with the given payload, and an ack function
+     */
+    static JmsMessage<?> of(javax.jms.Message msg, Executor executor, SessionMetadata sessionEntry) {
```

Comment:
method name should be `create`

Reply:
all factory methods should be `create` instead of `of`

Reply:
Trying to honor extended api of [Message](https://github.com/eclipse/microprofile-reactive-messaging/blob/master/api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java)